GameDataParser/Parsers/ExpTableParser.cs

```diff
@@ -1,29 +1,27 @@
-﻿using System;
-using System.Collections.Generic;
-using System.IO;
-using System.IO.MemoryMappedFiles;
+﻿using System.Collections.Generic;
 using System.Xml;
 using GameDataParser.Crypto.Common;
 using GameDataParser.Files;
 using Maple2Storage.Types.Metadata;
-using ProtoBuf;
 
 namespace GameDataParser.Parsers
 {
-    class ExpParser
+    class ExpTableParser : Exporter<List<ExpMetadata>>
     {
-        public static List<ExpMetadata> Parse(MemoryMappedFile m2dFile, IEnumerable<PackFileEntry> entries)
+        public ExpTableParser(MetadataResources resources) : base(resources, "exptable") { }
+
+        protected override List<ExpMetadata> parse()
```

Comment:
The metadata is still named `ExpMetadata`, we should choose to either name everything `ExpTable` or `Exp`

Reply:
The source already has this inconsistency, inside `ExpMetadataStorage` we have `ms2-exptable-metadata`. This is outside the scope of this PR

Reply:
Clearly it's not out of the scope because you renamed ExpParser to ExpTableParser and you removed ExpMetadataExport which is essentially renaming to use ExpTable as well since they are combined now.

Reply:
I renamed it here to match the name of the file, since this was the only file that didn't match the name it exported to. Also I removed all MetadataExport files, it had nothing to do with the name

Reply:
So renaming it to match the file for one thing is ok but not for the others, I see.

Reply:
Renaming this metadata affects the entire source, not just the parser, which is what this PR is for, how is this that difficult to understand? If I rename the metadata, then it affects the storage, then I have to rename that as well

Reply:
Ok then why didn't you just rename the file to `Exp` then instead of now having more files named `ExpTable`, before the only instance of `exptable` was in the metadata file name, now there are 2 files named that way and 2 files named the other way as opposed to before these changes it was 1 file named that way and 4 files named the other way.

Reply:
I changed 25 files this PR, all in GameDataParser only, the title of this PR: " GameDataParser architecture overhaul". I'm not making changes outside of the scope of this PR just because you feel like it. If the issue is that somebody named the file `exptable` but used `Exp` elsewhere, that's not really my problem or the point of this PR. Feel free to tag other developers, they will side with me, this PR has no business touching files that affect the base source

Reply:
Ok I will make a PR to fix the name + the extra name I have to change now because you changed it to ExpTable.